File: python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):
        self.rows = [list(map(int, row.split(' ')))  for row in matrix_string.split("\n")]

    def row(self, one_based_index):
        return self.rows[one_based_index-1]

    def column(self, one_based_index):
        return [row[one_based_index-1] for row in self.rows]

File: ruby/matrix/matrix.rb
# frozen_string_literal: true

class Matrix
  attr_reader :rows
  def initialize(string_representation)
    @rows = string_representation
            .each_line
            .map do |row|
              row.split.map(&:to_i)
            end
  end

  def columns
    rows.transpose
  end
end
